workflow/executor/docker/docker.go

```diff
@@ -86,6 +89,37 @@ func (d *DockerExecutor) GetOutputStream(containerID string, combinedOutput bool
 	return reader, nil
 }
 
+func (d *DockerExecutor) GetExitCode(containerID string) (int32, error) {
+	cmd := exec.Command("docker", "inspect", containerID, "--format='{{.State.ExitCode}}'")
+	log.Info(cmd.Args)
+	reader, err := cmd.StdoutPipe()
+	if err != nil {
+		return -1, errors.InternalWrapError(err)
+	}
+	err = cmd.Start()
+	if err != nil {
+		return -1, errors.InternalWrapError(err)
+	}
+	defer func() { _ = reader.Close() }()
+	bytes, err := ioutil.ReadAll(reader)
+	if err != nil {
+		return -1, errors.InternalWrapError(err)
+	}
+	out := string(bytes)
+
+	// Trims off a single newline for user convenience
+	outputLen := len(out)
+	if outputLen > 0 && out[outputLen-1] == '\n' {
+		out = out[0 : outputLen-1]
+	}
+	out = strings.Trim(out, `'`)
+	i, err := strconv.ParseInt(out, 10, 32)
```

Comment:
Curious -- why not use `strconv.Atoi(out)` here?

Reply:
Because it is the first time I code using Golang so I am basically lost. :p